### ISSUE TITLE
Is there anyway to have copy/pasting in this TextField work while still having this page styled the same way?

### ISSUE THREAD

**Issue by Pyroscout** [flutter/flutter#34013]
The `TextField` that I have on this page is not able to have any cut/copy/paste/select all actions done to it when you select text.

The options appear but when they are tapped nothing happened. I have determined that something is blocking the input and think I have found the source but it doesn't really make sense how the source I found is blocking it or how I can still have my page look the same while having this functionality.

I believe the source of the issue is 
```
Opacity(
    opacity: 1,
    child: Container(
        alignment: Alignment.bottomLeft,
        height: 90,
        padding: EdgeInsets.only(left: 20, right: 20),
        child: Text('Welcome to \nMy App'),
    ),
),
```

Below is a simplified version of my app recreating the issue.

```
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.white,
      body: Scaffold(
        resizeToAvoidBottomPadding: false,
        body: Container(
          color: Colors.white,
          child: ListView(children: <Widget>[
            Container(
              height: 50,
            ),
            Stack(
              children: <Widget>[
                Opacity(
                  opacity: 1,
                  child: Container(
                    alignment: Alignment.bottomLeft,
                    height: 90,
                    padding: EdgeInsets.only(left: 20, right: 20),
                    child: Text('Welcome to \nMy App'),
                  ),
                ),
                Opacity(
                  opacity: 0,
                  child: Container(),
                ),
              ],
            ),
            Container(height: 15,),
            Container(
              height: 2,
              alignment: Alignment.centerLeft,
              padding: EdgeInsets.only(left: 20, right: 20),
              child: FractionallySizedBox(
                widthFactor: 0.4,
                child: Container(color: Colors.grey,),
              ),
            ),
            Container(height: 15,),
            Container(
                height: 265,
                child: Stack(
                  children: <Widget>[
                    AnimatedOpacity(
                      opacity: 1,
                      duration: Duration(milliseconds: 200),
                      child: IgnorePointer(
                        ignoring: false,
                        child: Navigator(
                          onGenerateRoute: (RouteSettings settings) {
                            WidgetBuilder builder = (BuildContext _) => TextField();
                            return MaterialPageRoute(
                                builder: builder, settings: settings);
                          },
                        ),
                      ),
                    ),
                  ],
                )),
          ]),
        ),
      ),
    );
  }
}
```

When lines 44-52, the lines I put before I put the whole file, are commented out, cut/copy/paste/select all works as normal. When they are not, tapping any of these options causes nothing to happen.

Flutter Doctor
```
[✓] Flutter (Channel stable, v1.5.4-hotfix.2, on Mac OS X 10.14.5 18F132, locale en-US)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[✓] iOS toolchain - develop for iOS devices (Xcode 10.2.1)
[✓] Android Studio (version 3.4)
[✓] Connected device (1 available)
```

**Comment by kenthinson** [comment]
It annually looks like a bug with nested Navigators. Not anything to do with the opacity code you highlighted above. You can replace that code with any widget above the Navigator and it will be blocking the copy / cut / paste.

If I had to guess I would say flutter is "cutting out" space for the nested navigator and then inserting the navigator rendering into. Think like iframe.  Also guessing that the cut / copy / paste is handled by the system not flutter. So that is why it breaks out of this "sub window" visually, but the touches dont go through. Probably consumed by the widget that is technically "above" in the touches stack.

Long story short. For now probably need make some room above the text box or don't put it in a navigator for now. The flutter team will have to patch it.

**Comment by justinmc** [comment]
Below is an even more minimal app that reproduces the problem.  I agree it is something to do with Navigator.  It seems like anything extending outside the Navigator is not receiving events.  Hopefully someone that knows more about Navigator can shed some light, maybe @goderbauer?

By the way, the selection menu is actually rendered by Flutter just like everything else.  It uses an Overlay, but same idea to what you were thinking.  See [text_selection.dart](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/text_selection.dart).

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: Scaffold(
        appBar: AppBar(
          title: Text('Flutter demo'),
        ),
        body: Center(
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: <Widget>[
              Container(
                height: 200,
                color: Colors.blue.withOpacity(0.5),
                child: Navigator(
                  onGenerateRoute: (RouteSettings settings) {
                    WidgetBuilder builder = (BuildContext _) => TextField();
                    return MaterialPageRoute(
                        builder: builder, settings: settings);
                  },
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}
```

**Comment by Pyroscout** [comment]
> Long story short. For now probably need make some room above the text box or don't put it in a navigator. The flutter team will have to patch it.

When you say "make some room above the text box" do you mean make room inside or outside the second navigator?

**Comment by justinmc** [comment]
My impression is that space above the TextField but within the Navigator is what would do it.  So in my example, you need to keep the selection menu inside the blue Navigator in order for it to be interactive.

<img width="459" alt="Screen Shot 2019-06-07 at 2 14 41 PM" src="https://user-images.githubusercontent.com/389558/59133997-a1f1f380-892e-11e9-8abd-b3e876d26715.png">

**Comment by kenthinson** [comment]
@justinmc That's interesting info! I haven't dove into the text_selection code. Like I was saying just was guessing at the problem :)

@Pyroscout Yes you will need to make more room inside the navigator view above the textfield for the cut / copy / paste. Or just don't have the text box inside a sub navigator. Depends on what your app is trying to do if that's possible or not.

**Comment by Pyroscout** [comment]
@justinmc @kenthinson I tried making space inside the navigator and it still is not letting me paste. Do you have an example where you made space and it worked?

Here is my code:
```import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.white,
      body: Scaffold(
        resizeToAvoidBottomPadding: false,
        body: Container(
          color: Colors.white,
          child: ListView(children: <Widget>[
            Container(
              height: 50,
            ),
            Stack(
              children: <Widget>[
                Opacity(
                  opacity: 1,
                  child: Container(
                    alignment: Alignment.bottomLeft,
                    height: 90,
                    padding: EdgeInsets.only(left: 20, right: 20),
                    child: Text('Welcome to \nMy App'),
                  ),
                ),
                Opacity(
                  opacity: 0,
                  child: Container(),
                ),
              ],
            ),
            Container(height: 15,),
            Container(
              height: 2,
              alignment: Alignment.centerLeft,
              padding: EdgeInsets.only(left: 20, right: 20),
              child: FractionallySizedBox(
                widthFactor: 0.4,
                child: Container(color: Colors.grey,),
              ),
            ),
            Container(height: 15,),
            Container(
                height: 265,
                child: Stack(
                  children: <Widget>[
                    AnimatedOpacity(
                      opacity: 1,
                      duration: Duration(milliseconds: 200),
                      child: IgnorePointer(
                        ignoring: false,
                        child: Navigator(
                          onGenerateRoute: (RouteSettings settings) {
                            WidgetBuilder builder = (BuildContext _) {
                              return Column(
                                children: <Widget>[
                                  Container(
                                    height: 100,
                                  ),
                                  TextField(),
                                ],
                              );
                            };
                            return MaterialPageRoute(
                                builder: builder, settings: settings);
                          },
                        ),
                      ),
                    ),
                  ],
                )),
          ]),
        ),
      ),
    );
  }
}```

**Comment by justinmc** [comment]
@Pyroscout I think I was too confident in my guess, sorry.  I tried it and got some weird behavior, but couldn't get a selection menu that always worked with various types of space at the top of the Navigator.  There might be a workaround somehow, but it looks like we definitely need a fix in the Flutter repo.

**Comment by Pyroscout** [comment]
@justinmc any idea when a fix for this might be implemented? Is there any way to find out or is it more likely that it won't be for a while?

**Comment by Pyroscout** [comment]
I actually found a workaround where if I increase the height of the container containing the navigator, a line that appears while Debug Paint is on and the cut/copy/paste menu is showing moves. Through a bit of testing I have found that this line is the separator of no return with being able to paste as long as there is also space above the TextField inside the navigator. I realize this explanation probably doesn't make much sense so I am gonna also explain with code and images.

Here is my file with working pasting:
```import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.white,
      body: Scaffold(
        resizeToAvoidBottomPadding: false,
        body: Container(
          color: Colors.white,
          child: ListView(children: <Widget>[
            Container(
              height: 50,
            ),
            Stack(
              children: <Widget>[
                Opacity(
                  opacity: 1,
                  child: Container(
                    alignment: Alignment.bottomLeft,
                    height: 90,
                    padding: EdgeInsets.only(left: 20, right: 20),
                    child: Text('Welcome to \nMy App'),
                  ),
                ),
                Opacity(
                  opacity: 0,
                  child: Container(),
                ),
              ],
            ),
            Container(height: 15,),
            Container(
              height: 2,
              alignment: Alignment.centerLeft,
              padding: EdgeInsets.only(left: 20, right: 20),
              child: FractionallySizedBox(
                widthFactor: 0.4,
                child: Container(color: Colors.grey,),
              ),
            ),
            Container(height: 15,),
            Container(
                height: 350,
                child: Stack(
                  children: <Widget>[
                    AnimatedOpacity(
                      opacity: 1,
                      duration: Duration(milliseconds: 200),
                      child: IgnorePointer(
                        ignoring: false,
                        child: Navigator(
                          onGenerateRoute: (RouteSettings settings) {
                            WidgetBuilder builder = (BuildContext _) {
                              return Column(
                                children: <Widget>[
                                  Container(
                                    height: 100,
                                  ),
                                  TextField(),
                                ],
                              );
                            };
                            return MaterialPageRoute(
                                builder: builder, settings: settings);
                          },
                        ),
                      ),
                    ),
                  ],
                )),
          ]),
        ),
      ),
    );
  }
}```

The difference between it and the last version is the height of the very bottom container inside the list view which I increased to 350.

Circled in red is the location of the line
<img width="378" alt="Screen Shot 2019-06-12 at 12 14 50 PM" src="https://user-images.githubusercontent.com/4473004/59375783-1f8d7900-8d0c-11e9-9650-91e42659402c.png">

Compared to this image without it, the previous image has a double set of lines rather than a single set
<img width="381" alt="Screen Shot 2019-06-12 at 12 14 37 PM" src="https://user-images.githubusercontent.com/4473004/59375678-e94ff980-8d0b-11e9-8b10-206481f1ffc6.png">

This final image shows where the line was before I increased the height of the container, which was above the paste box and was, to my best understanding, why the paste box is unable to be selected.
<img width="383" alt="Screen Shot 2019-06-12 at 12 19 30 PM" src="https://user-images.githubusercontent.com/4473004/59376027-962a7680-8d0c-11e9-99c5-efa589a59a36.png">

**Comment by justinmc** [comment]
@Pyroscout I really appreciate you posting back with a detailed explanation of the workaround!  Should save people a lot of time if they run into this.  Something was not tall enough to enclose the selection menu.  Either it should be by default, or the selection menu should be interactive event outside of it.

**Comment by justinmc** [comment]
I tried my repro steps from https://github.com/flutter/flutter/issues/34013#issuecomment-499952438 and I don't see the problem anymore.  If anyone else is still experiencing this, please post and we can reopen this!
